operator/consumer/event_listener.go

```diff
@@ -0,0 +1,59 @@
+package consumer
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+type EventListener interface {
+	OnArrival()
+	OnFormatError()
+}
+
+const OperatorNamespace = "sffl_operator"
+const ConsumerSubsystem = "consumer"
+
+type SelectiveListener struct {
+	OnArrivalCb     func()
+	OnFormatErrorCb func()
+}
+
+func (l *SelectiveListener) OnArrival() {
+	if l.OnArrivalCb != nil {
+		l.OnArrivalCb()
+	}
+}
+
+func (l *SelectiveListener) OnFormatError() {
+	if l.OnFormatErrorCb != nil {
+		l.OnFormatErrorCb()
+	}
+}
+
+func MakeConsumerMetrics(registry *prometheus.Registry) EventListener {
```

Comment:
Same comment from `MakeAttestorMetrics` applies here (and in all other metrics registering)